File: modified_scripts/julia/binary_search/qwen2.5-coder_7b.jl
```julia
function binary_search(arr::Vector{T}, x::T) where T
    low = 1
    high = length(arr)
    while low <= high
        mid = div(low + high, 2)
        if arr[mid] == x
            return mid
        elseif arr[mid] < x
            low = mid + 1
        else
            high = mid - 1
        end
    end
    return -1
end
```

File: modified_scripts/r/binary_search/qwen2.5-coder_7b.r

binary_search <- function(arr, x) {
  low <- 1
  high <- length(arr)
  while (low <= high) {
    mid <- floor((low + high) / 2)
    if (arr[mid] == x) return(mid)
    else if (arr[mid] < x) low <- mid + 1
    else high <- mid - 1
  }
  return(-1)
}
